src/operator/tensor/elemwise_unary_op.h

```diff
@@ -457,30 +457,67 @@ class UnaryOp : public OpBase {
 };
 
 #if MXNET_USE_ONEDNN == 1
-inline bool TanhStorageType(const nnvm::NodeAttrs& attrs,
-                            const int dev_mask,
-                            DispatchMode* dispatch_mode,
-                            std::vector<int>* in_attrs,
-                            std::vector<int>* out_attrs) {
+inline bool EltwiseStorageType(const nnvm::NodeAttrs& attrs,
+                               const int dev_mask,
+                               DispatchMode* dispatch_mode,
+                               std::vector<int>* in_attrs,
+                               std::vector<int>* out_attrs) {
   CHECK_EQ(in_attrs->size(), 1);
   CHECK_EQ(out_attrs->size(), 1);
 
   return DNNLStorageType(attrs, dev_mask, true, dispatch_mode, in_attrs, out_attrs);
 }
 
-inline void TanhComputeExCPU(const nnvm::NodeAttrs& attrs,
-                             const OpContext& ctx,
-                             const std::vector<mxnet::NDArray>& inputs,
-                             const std::vector<OpReqType>& req,
-                             const std::vector<mxnet::NDArray>& outputs) {
-  if (SupportDNNLTanh(inputs[0], outputs[0])) {
+template <typename OP>
+struct DNNLAlgorithm {};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::plus> {
+  static const dnnl::algorithm value = dnnl::algorithm::binary_add;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::minus> {
+  static const dnnl::algorithm value = dnnl::algorithm::binary_sub;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::mul> {
+  static const dnnl::algorithm value = dnnl::algorithm::binary_mul;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::div> {
+  static const dnnl::algorithm value = dnnl::algorithm::binary_div;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::tanh> {
+  static const dnnl::algorithm value = dnnl::algorithm::eltwise_tanh;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::exp> {
+  static const dnnl::algorithm value = dnnl::algorithm::eltwise_exp;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::square> {
+  static const dnnl::algorithm value = dnnl::algorithm::eltwise_square;
+};
+template <>
+struct DNNLAlgorithm<op::mshadow_op::square_root> {
+  static const dnnl::algorithm value = dnnl::algorithm::eltwise_sqrt;
+};
```

Comment:
I believe we can turn `DNNLAlgorithm` into a constexpr function which would be more readable imo.

Reply:
op::mshadow_op::XXX is not a literal_type afaik.

Reply:
I thought `op::mshadow_op` is an enum, like dnnl::algorithm xd. We could still turn this into a function, but it wouldn't be that more readable anymore.

Reply:
I like it as is :)

Reply:
Here is the function if you are intrerested:
```c++
template <class>
inline constexpr bool dependent_false_v = false;
template <typename MXNetOP>
constexpr dnnl::algorithm DNNLAlgorithm() {
  if constexpr (std::is_same_v<MXNetOP, op::mshadow_op::plus>) {
    return dnnl::algorithm::binary_add;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::minus>) {
    return dnnl::algorithm::binary_sub;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::mul>) {
    return dnnl::algorithm::binary_mul;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::div>) {
    return dnnl::algorithm::binary_div;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::tanh>) {
    return dnnl::algorithm::eltwise_tanh;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::exp>) {
    return dnnl::algorithm::eltwise_exp;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::square>) {
    return dnnl::algorithm::eltwise_square;
  } else if (std::is_same_v<MXNetOP, op::mshadow_op::square_root>) {
    return dnnl::algorithm::eltwise_sqrt;
  } else {
    static_assert(dependent_false_v<MXNetOP>, "Error");
    return dnnl::algorithm();
  }
}
```